### PR TITLE
Define separate conditions for parsing errors

### PR DESCRIPTION
**Pull request by adlai**
There are also a handful of minor nuisances in the same commit;
and merging this should close Issue #18

### CODE REVIEW

**Comment by mbattyani** [comment]
Excellent! Thanks

**Comment by adlai** [comment]
Thank you for merging the PR, @mbattyani ; it appears that I didn't get the GitHub metadata correct, so it did not automatically close Issue #18 that can probably be closed now.

File: change-log.txt [hunk 1]
```diff
@@ -1,19 +1,23 @@
 ;;; cl-pdf copyright 2002-2004 Marc Battyani see license.txt for details of the license
 ;;; You can reach me at marc.battyani@fractalconcept.com or marc@battyani.net
 
-This is the change log for cl-pdf. 
+This is the change log for cl-pdf.
 See also contributors.txt and to-do.txt
 
 Current version:
+   This release supports features in the PDF 1.4 specification
```
Comment (adlai) [inline change-log.txt:8]: Sometimes, I read text I wrote and get pissed off at myself...

Let's hope nobody expects the word "all" to magically appear.

File: pdf-parser.lisp [hunk 2]
```diff
@@ -45,35 +45,59 @@
...
-;;; Parser
-;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Parsing Errors
 
-(define-condition pdf-parse-error ()
+(define-condition pdf-parse-error (parse-error)
   ((stream :initarg :stream :reader pdf-parse-error-stream)
-   (message :initarg :message :reader pdf-parse-error-message))
-  (:report (lambda (condition stream)
-             (format stream "~&Error at position ~D while reading PDF document.~%~A~%~A~%"                     (file-position (pdf-parse-error-stream condition))                     (pdf-parse-error-message condition)
```
Comment (adlai) [inline pdf-parser.lisp:56]:
```
It is obnoxious to complain at myself for not documenting
all the different kinds of NOP performed by my code, yet
big business and nobody claims prettification has no hot
head sun shade neither. Code as you would professionally!
```

**Review by adlai — COMMENTED** [review]
Hello, it looks like you're re-reading me talking at myself; perhaps you'd like to shill for Microsoft?

**Comment by mbattyani** [comment]
Hey Adlai,
I think that text is clear enough but feel free to change it if you prefer

**Pull request merged**